lib/edvs.js

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const brHttpsAgent = require('bedrock-https-agent');
+const {EdvClient} = require('edv-client');
+const keyResolver = require('./keyResolver');
+const kms = require('./kms');
+
+// load config defaults
+require('./config');
+const KMS_MODULE = 'ssm-v1';
+
+// module API
+const api = {};
+module.exports = api;
+
+/**
+ * Creates an EdvClient
+ *
+ *
+ * @return {Promise<EdvClient>} resolves to a EdvClient.
+ */
+api.create = async (
+  {invocationSigner, kmsClient, referenceId, profileId} = {}) => {
+  const edvBaseUrl = `${bedrock.config.server.baseUri}/edvs`;
+  const [keyAgreementKey, hmac] = await Promise.all([
+    kms.generateKey({
+      invocationSigner,
+      type: 'keyAgreement',
+      kmsClient,
+      kmsModule: KMS_MODULE
+    }),
+    kms.generateKey({
+      invocationSigner,
+      type: 'hmac',
+      kmsClient,
+      kmsModule: KMS_MODULE
+    })
+  ]);
+  // create edv
+  let config = {
+    sequence: 0,
+    controller: profileId,
+    keyAgreementKey: {id: keyAgreementKey.id, type: keyAgreementKey.type},
+    hmac: {id: hmac.id, type: hmac.type}
+  };
+  if(referenceId) {
+    config.referenceId = referenceId;
+  }
+  const {httpsAgent} = brHttpsAgent;
+  const capability = `${edvBaseUrl}/zcaps/configs`;
+  // TODO: Update to use the bedrock-edv-storage NodeJS API
```

Comment:
I think we should expect that we want to reuse this module to be able to create EDVs on a non-local EDV service. If we were to use the bedrock-edv-storage NodeJS API then this code would not work for that use case. We might still need to modify this function a bit to make it work for non-local services, but we should be careful not to make it even more difficult to do so without good reason.

Reply:
Added a TODO.